Doc/reference/expressions.rst

```diff
@@ -183,8 +183,20 @@ by considering each of the :keyword:`for` or :keyword:`if` clauses a block,
 nesting from left to right, and evaluating the expression to produce an element
 each time the innermost block is reached.
 
-Note that the comprehension is executed in a separate scope, so names assigned
-to in the target list don't "leak" into the enclosing scope.
+However, aside from the leftmost :keyword:`for` clause, the comprehension is
```

Comment:
I think "leftmost `for` clause" is still ambiguous. Does this include the <expr> in `for a[<expr>] in ...`? It should not, but I would assume it's part of "the `for` clause." Maybe say "the iterable in the leftmost `for` clause"?

Reply:
Using just "iterable" ended up looking odd in context (since executing an iterable could be taken to mean iterating over it).

However, "the iterable expression in the leftmost :keyword:`for` clause" reads reasonably well to me, and I also made it explicit that the leftmost filter condition is evaluated in the nested scope.

It may be worth our while to explicitly define "outermost iterable" as a formal term in the glossary, but I think that would be better handled as a distinct docs RFE.